ui/hooks/useDisplayName.ts

```diff
@@ -37,11 +47,23 @@ export function useDisplayNames(
 
   const watchedNftNames = useSelector(getNftContractsByAddressOnCurrentChain);
 
-  return requests.map(({ value, preferContractSymbol }, index) => {
+  return requests.map(({ value, preferContractSymbol, tokenId }, index) => {
     const nameEntry = nameEntries[index];
     const firstPartyContractName = firstPartyContractNames[index];
     const singleContractInfo = contractInfo[index];
     const watchedNftName = watchedNftNames[value.toLowerCase()]?.name;
+    const nftCollectionProperties =
+      nftCollections[
+        `${value.toLowerCase()}:${hexToDecimal(tokenId as string)}`
```

Comment:
This ultimately shouldn't be necessary as the API can source whether a contract address `isSpam` from the address alone, I've noted this in the core PR also.

Reply:
Done